test/integration/machine-learning.spec.ts

```diff
@@ -14,9 +14,117 @@
  * limitations under the License.
  */
 
+
+import path = require('path');
+import * as chai from 'chai';
 import * as admin from '../../lib/index';
+import {Bucket} from '@google-cloud/storage';
+
+const expect = chai.expect;
 
 describe('admin.machineLearning', () => {
+
+  const modelsToDelete: string[] = [];
+
+  function scheduleForDelete(model: admin.machineLearning.Model) {
+    modelsToDelete.push(model.modelId);
+  }
+
+  function unscheduleForDelete(model: admin.machineLearning.Model) {
+    modelsToDelete.splice(modelsToDelete.indexOf(model.modelId), 1);
+  }
+
+  function deleteTempModels(): Promise<void[]> {
+    const promises: Array<Promise<void>> = [];
+    modelsToDelete.forEach((modelId) => {
+      promises.push(admin.machineLearning().deleteModel(modelId));
+    });
+    modelsToDelete.splice(0, modelsToDelete.length);  // Clear out the array.
+    return Promise.all(promises);
+  }
+
+  function createTemporaryModel(options?: admin.machineLearning.ModelOptions)
+    : Promise<admin.machineLearning.Model> {
+    let modelOptions: admin.machineLearning.ModelOptions = {
+      displayName: 'nodejs_integration_temp_model',
+    };
+    if (options) {
+      modelOptions = options;
+    }
+    return admin.machineLearning().createModel(modelOptions)
+      .then((model) => {
+        scheduleForDelete(model);
+        return model;
+      });
+  }
+
+  function uploadModelToGcs(localFileName: string, gcsFileName: string): string {
```

Comment:
This should return a Promise. Method implementation should handle the Promise returned by upload correctly.

Reply:
done